app/src/main/java/com/marcelo/marvelheroes/presentation/ui/fragments/sort/SortHeroesFragment.kt

```diff
@@ -0,0 +1,148 @@
+                handleSaveOrderHeroes()
+            }
+        }
+    }
+
+    private fun getOrderByValue(chipId: Int): String = when (chipId) {
+        R.id.chipOrderName -> ORDER_BY_NAME_ASCENDING
+        R.id.chipOrderModified -> ORDER_BY_MODIFIED_ASCENDING
+        else -> ORDER_BY_NAME_ASCENDING
+    }
+
+    private fun getOrderValue(chipId: Int): String = when (chipId) {
+        R.id.chipOrderGrowing -> ASCENDING
+        R.id.chipOrderDescending -> DESCENDING
+        else -> ASCENDING
+    }
+
+    private fun fetchSaveOrderHeroes() {
+        viewModel.saveOrderHeroes(orderBy, order)
+    }
+
+    private fun handleGetOrderHeroes() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.CREATED) {
+                viewModel.viewStateGetSortHeroes.collect { state ->
+                    when (state) {
+                        is State.Loading -> {
+                            binding.btnApplySort.visibility = View.GONE
+                            binding.progressBarSort.visibility = View.VISIBLE
+                        }
+
+                        is State.Success -> {
+                            binding.btnApplySort.visibility = View.VISIBLE
+                            binding.progressBarSort.visibility = View.GONE
+                            if (state.data is SortHeroesViewData.Success) {
+                                successGetOrderHeroes(state.data)
+                            }
+                        }
+
+                        else -> {}
+                    }
+                }
+            }
+        }
+    }
+
+    private fun successGetOrderHeroes(data: SortHeroesViewData.Success) {
+        with(binding) {
+            val (orderBy, orderOptions) = data.sortingPair
+
+            chipGroupOrderByTitle.forEach { chip ->
+                if (getOrderByValue(chip.id) == orderBy) {
+                    (chip as Chip).isChecked = true
+                }
+            }
+
+            chipGroupOrderByOptions.forEach { chip ->
+                if (getOrderValue(chip.id) == orderOptions) {
+                    (chip as Chip).isChecked = true
+                }
+            }
+        }
+    }
+
+    private fun handleSaveOrderHeroes() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.CREATED) {
+                viewModel.viewStateSaveSortHeroes.collect { state ->
+                    when (state) {
+                        is State.Loading -> {
+                            binding.btnApplySort.visibility = View.GONE
+                            binding.progressBarSort.visibility = View.VISIBLE
+                        }
+
+                        is State.Success -> {
+                            binding.btnApplySort.visibility = View.VISIBLE
+                            binding.progressBarSort.visibility = View.GONE
+                        }
+
+                        else -> {}
```

Comment:
remove esse else. A grande vantagem de modelar states com sealed classe é ser forcado a cobrir todas as possibilidades. Se deixar o else e outra status for adicionado, o when nao vai reclamar.